src/computational_graph/transform.jl

```diff
@@ -87,21 +81,121 @@ end
 """
     function standardize_labels!(g::Graph)
 
-This function first finds all labels involved in g and its subgraphs (for example, 1, 4, 5, 7, ...), 
-then relabel them in the order 1, 2, 3, 4, ....
-Return a copy of stantardized g.
+    This function first finds all labels involved in g and its subgraphs (for example, 1, 4, 5, 7, ...), 
+    then relabel them in the order 1, 2, 3, 4, ....
+    Returns a standardized copy of g.
 """
 standardize_labels(g::Graph) = standardize_labels!(deepcopy(g))
 
-############LEGACY BELOW################
+"""
+    function replace_subgraph!(g::Graph, w::Graph, m::graph)
 
-# function readDiag(io::IO)
+    In place function that replaces the children graph w in graph g with a new graph m.
+    Graphs w and m should have the same internal and external vertices, and topology
+"""
+function replace_subgraph!(g::Graph, w::Graph, m::Graph)
+    @assert !isleaf(g) "Target parent graph can not be a leaf"
+    @assert w.vertices == m.vertices "Old and new subgraph should have the same vertices"
```

Comment:
1. why replacing subgraph requires vertices to be the same? I think you can replace the subgraph as long as the external legs are the same?
2. It is also important that the Graph.order is the same

Reply:
Resolved